# Accenture Data Analyst Project

In [1]:
import pandas as pd
import numpy as np

## Reactions

In [2]:
df = pd.read_csv('Reactions.csv')
df.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  97522e57-d9ab-4bd6-97bf-c24d952602d2   
2           2  97522e57-d9ab-4bd6-97bf-c24d952602d2   
3           3  97522e57-d9ab-4bd6-97bf-c24d952602d2   
4           4  97522e57-d9ab-4bd6-97bf-c24d952602d2   

                                User ID     Type             Datetime  
0                                   NaN      NaN  2021-04-22 15:17:15  
1  5d454588-283d-459d-915d-c48a2cb4c27f  disgust  2020-11-07 09:43:50  
2  92b87fa5-f271-43e0-af66-84fac21052e6  dislike  2021-06-17 12:22:51  
3  163daa38-8b77-48c9-9af6-37a6c1447ac2   scared  2021-04-18 05:13:58  
4  34e8add9-0206-47fd-a501-037b994650a2  disgust  2021-01-06 19:13:01

In [3]:
df.shape

(25553, 5)

In [4]:
# Removing unwanted column

df.drop(columns = ['Unnamed: 0','User ID'], inplace = True)
df.head()

Content ID     Type             Datetime
0  97522e57-d9ab-4bd6-97bf-c24d952602d2      NaN  2021-04-22 15:17:15
1  97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2020-11-07 09:43:50
2  97522e57-d9ab-4bd6-97bf-c24d952602d2  dislike  2021-06-17 12:22:51
3  97522e57-d9ab-4bd6-97bf-c24d952602d2   scared  2021-04-18 05:13:58
4  97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2021-01-06 19:13:01

In [5]:
# Checking for null values

df.isna().sum()

Content ID      0
Type          980
Datetime        0
dtype: int64

In [6]:
# removing null values

df = df.dropna()
df.head()

Content ID        Type             Datetime
1  97522e57-d9ab-4bd6-97bf-c24d952602d2     disgust  2020-11-07 09:43:50
2  97522e57-d9ab-4bd6-97bf-c24d952602d2     dislike  2021-06-17 12:22:51
3  97522e57-d9ab-4bd6-97bf-c24d952602d2      scared  2021-04-18 05:13:58
4  97522e57-d9ab-4bd6-97bf-c24d952602d2     disgust  2021-01-06 19:13:01
5  97522e57-d9ab-4bd6-97bf-c24d952602d2  interested  2020-08-23 12:25:58

In [7]:
df.shape

(24573, 3)

In [8]:
Reaction = df

## Reaction Types

In [9]:
df = pd.read_csv('ReactionTypes.csv')
df.head()

Unnamed: 0        Type Sentiment  Score
0           0       heart  positive     60
1           1        want  positive     70
2           2     disgust  negative      0
3           3        hate  negative      5
4           4  interested  positive     30

In [10]:
df = df.drop(columns = 'Unnamed: 0')
df.head()

Type Sentiment  Score
0       heart  positive     60
1        want  positive     70
2     disgust  negative      0
3        hate  negative      5
4  interested  positive     30

In [11]:
df.shape

(16, 3)

In [12]:
ReactionTypes = df

## Content

In [13]:
df = pd.read_csv('Content.csv')
df.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
2           2  230c4e4d-70c3-461d-b42c-ec09396efb3f   
3           3  356fff80-da4d-4785-9f43-bc1261031dc6   
4           4  01ab84dd-6364-4236-abbb-3f237db77180   

                                User ID   Type        Category  \
0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo        Studying   
1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo  healthy eating   
2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo  healthy eating   
3  9fb4ce88-fac1-406c-8544-1a899cee7aaf  photo      technology   
4  e206e31b-5f85-4964-b6ea-d7ee5324def1  video            food   

                                                 URL  
0  https://socialbuzz.cdn.com/content/storage/975...  
1  https://socialbuzz.cdn.com/content/storage/9f7...  
2  https://socialbuzz.cdn.com/content/storage/230...  
3  https://socialbuzz.cdn.com/content/storage/356...  
4  https://socialbuzz.cdn.com/content/storage/01a...

In [14]:
df.shape

(1000, 6)

In [15]:
df = df.drop(columns = ['Unnamed: 0','User ID','URL'])
df.head()

Content ID   Type        Category
0  97522e57-d9ab-4bd6-97bf-c24d952602d2  photo        Studying
1  9f737e0a-3cdd-4d29-9d24-753f4e3be810  photo  healthy eating
2  230c4e4d-70c3-461d-b42c-ec09396efb3f  photo  healthy eating
3  356fff80-da4d-4785-9f43-bc1261031dc6  photo      technology
4  01ab84dd-6364-4236-abbb-3f237db77180  video            food

In [16]:
df.isna().sum()

Content ID    0
Type          0
Category      0
dtype: int64

In [17]:
df.shape

(1000, 3)

In [18]:
# Removing quotation marks in some values of category column

df['Category'] = df['Category'].str.strip(' " ')

In [19]:
Content = df
Content.rename(columns = {'Type':'Content Type'}, inplace = True)
Content.head()

Content ID Content Type        Category
0  97522e57-d9ab-4bd6-97bf-c24d952602d2        photo        Studying
1  9f737e0a-3cdd-4d29-9d24-753f4e3be810        photo  healthy eating
2  230c4e4d-70c3-461d-b42c-ec09396efb3f        photo  healthy eating
3  356fff80-da4d-4785-9f43-bc1261031dc6        photo      technology
4  01ab84dd-6364-4236-abbb-3f237db77180        video            food

In [20]:
Reaction.head()

Content ID        Type             Datetime
1  97522e57-d9ab-4bd6-97bf-c24d952602d2     disgust  2020-11-07 09:43:50
2  97522e57-d9ab-4bd6-97bf-c24d952602d2     dislike  2021-06-17 12:22:51
3  97522e57-d9ab-4bd6-97bf-c24d952602d2      scared  2021-04-18 05:13:58
4  97522e57-d9ab-4bd6-97bf-c24d952602d2     disgust  2021-01-06 19:13:01
5  97522e57-d9ab-4bd6-97bf-c24d952602d2  interested  2020-08-23 12:25:58

In [21]:
ReactionTypes.head()

Type Sentiment  Score
0       heart  positive     60
1        want  positive     70
2     disgust  negative      0
3        hate  negative      5
4  interested  positive     30

In [22]:
# Merging table Reaction and Content

Table = pd.merge(Reaction, Content, on = 'Content ID')
Table

Content ID        Type             Datetime  \
0      97522e57-d9ab-4bd6-97bf-c24d952602d2     disgust  2020-11-07 09:43:50   
1      97522e57-d9ab-4bd6-97bf-c24d952602d2     dislike  2021-06-17 12:22:51   
2      97522e57-d9ab-4bd6-97bf-c24d952602d2      scared  2021-04-18 05:13:58   
3      97522e57-d9ab-4bd6-97bf-c24d952602d2     disgust  2021-01-06 19:13:01   
4      97522e57-d9ab-4bd6-97bf-c24d952602d2  interested  2020-08-23 12:25:58   
...                                     ...         ...                  ...   
24568  75d6b589-7fae-4a6d-b0d0-752845150e56     dislike  2020-06-27 09:46:48   
24569  75d6b589-7fae-4a6d-b0d0-752845150e56   intrigued  2021-02-16 17:17:02   
24570  75d6b589-7fae-4a6d-b0d0-752845150e56  interested  2020-09-12 03:54:58   
24571  75d6b589-7fae-4a6d-b0d0-752845150e56     worried  2020-11-04 20:08:31   
24572  75d6b589-7fae-4a6d-b0d0-752845150e56     cherish  2021-01-04 04:55:11   

      Content Type    Category  
0            photo    Studying  
1            photo    Studying  
2            photo    Studying  
3            photo    Studying  
4            photo    Studying  
...            ...         ...  
24568        audio  technology  
24569        audio  technology  
24570        audio  technology  
24571        audio  technology  
24572        audio  technology  

[24573 rows x 5 columns]

In [23]:
# Merging Final Table and Reaction Types

Table = pd.merge(Table, ReactionTypes, on = 'Type')
Table

Content ID     Type             Datetime  \
0      97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2020-11-07 09:43:50   
1      97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2021-01-06 19:13:01   
2      97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2021-04-09 02:46:20   
3      9f737e0a-3cdd-4d29-9d24-753f4e3be810  disgust  2021-03-28 21:15:26   
4      230c4e4d-70c3-461d-b42c-ec09396efb3f  disgust  2020-08-04 05:40:33   
...                                     ...      ...                  ...   
24568  435007a5-6261-4d8b-b0a4-55fdc189754b    adore  2020-10-04 22:26:33   
24569  435007a5-6261-4d8b-b0a4-55fdc189754b    adore  2020-09-18 10:50:50   
24570  4e4c9690-c013-4ee7-9e66-943d8cbd27b7    adore  2020-10-31 03:58:44   
24571  4e4c9690-c013-4ee7-9e66-943d8cbd27b7    adore  2020-06-25 15:12:29   
24572  4e4c9690-c013-4ee7-9e66-943d8cbd27b7    adore  2020-12-17 16:32:57   

      Content Type        Category Sentiment  Score  
0            photo        Studying  negative      0  
1            photo        Studying  negative      0  
2            photo        Studying  negative      0  
3            photo  healthy eating  negative      0  
4            photo  healthy eating  negative      0  
...            ...             ...       ...    ...  
24568        audio        veganism  positive     72  
24569        audio        veganism  positive     72  
24570          GIF         culture  positive     72  
24571          GIF         culture  positive     72  
24572          GIF         culture  positive     72  

[24573 rows x 7 columns]

In [24]:
Table['Category'] = Table['Category'].str.capitalize()
Table.head()

Content ID     Type             Datetime  \
0  97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2020-11-07 09:43:50   
1  97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2021-01-06 19:13:01   
2  97522e57-d9ab-4bd6-97bf-c24d952602d2  disgust  2021-04-09 02:46:20   
3  9f737e0a-3cdd-4d29-9d24-753f4e3be810  disgust  2021-03-28 21:15:26   
4  230c4e4d-70c3-461d-b42c-ec09396efb3f  disgust  2020-08-04 05:40:33   

  Content Type        Category Sentiment  Score  
0        photo        Studying  negative      0  
1        photo        Studying  negative      0  
2        photo        Studying  negative      0  
3        photo  Healthy eating  negative      0  
4        photo  Healthy eating  negative      0

In [25]:
df = Table[['Category','Score']]
df

Category  Score
0            Studying      0
1            Studying      0
2            Studying      0
3      Healthy eating      0
4      Healthy eating      0
...               ...    ...
24568        Veganism     72
24569        Veganism     72
24570         Culture     72
24571         Culture     72
24572         Culture     72

[24573 rows x 2 columns]

In [26]:
Score = df.groupby('Category')['Score'].sum()
Score

Category
Animals            74965
Cooking            64756
Culture            66579
Dogs               52511
Education          57436
Fitness            55323
Food               66676
Healthy eating     69339
Public speaking    49264
Science            71168
Soccer             57783
Studying           54269
Technology         68738
Tennis             50339
Travel             64880
Veganism           49619
Name: Score, dtype: int64

In [27]:
Score.sort_values(ascending = False)

Category
Animals            74965
Science            71168
Healthy eating     69339
Technology         68738
Food               66676
Culture            66579
Travel             64880
Cooking            64756
Soccer             57783
Education          57436
Fitness            55323
Studying           54269
Dogs               52511
Tennis             50339
Veganism           49619
Public speaking    49264
Name: Score, dtype: int64

### The Top 5 categories are -

- Animals         =     74965
- Science         =     71168
- Healthy eating  =     69339
- Technology      =     68738
- Food            =     66676

In [28]:
# Table.to_excel('cleaned_dataset.xlsx', index=False)